src/test/java/com/commercetools/sync/commons/utils/ResourceCustomUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,569 @@
+        assertThat(setCustomFieldsUpdateActions).hasSize(2);
+        final UpdateAction<Category> categoryUpdateAction = setCustomFieldsUpdateActions.get(0);
+        assertThat(categoryUpdateAction).isNotNull();
+        assertThat(categoryUpdateAction).isInstanceOf(SetCustomField.class);
+    }
+
+    @Test
+    public void buildSetCustomFieldsUpdateActions_WithNoNewCustomFieldsInOldCustomFields_ShouldBuildUpdateActions() {
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+
+        final Map<String, JsonNode> newCustomFields = new HashMap<>();
+        newCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+        newCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot"));
+        newCustomFields.put("url", JsonNodeFactory.instance.objectNode().put("domain", "domain.com"));
+        newCustomFields.put("size", JsonNodeFactory.instance.objectNode().put("cm", 34));
+
+        final List<UpdateAction<Category>> setCustomFieldsUpdateActions =
+            buildSetCustomFieldsUpdateActions(oldCustomFields, newCustomFields, mock(Category.class),
+                new CategoryCustomActionBuilder(),null, category -> null);
+
+        assertThat(setCustomFieldsUpdateActions).isNotNull();
+        assertThat(setCustomFieldsUpdateActions).isNotEmpty();
+        assertThat(setCustomFieldsUpdateActions).hasSize(4);
+    }
+
+    @Test
+    public void buildSetCustomFieldsUpdateActions_WithOldCustomFieldNotInNewFields_ShouldBuildUpdateActions() {
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+        oldCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+        oldCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot").put("en", "red"));
+
+        final Map<String, JsonNode> newCustomFields = new HashMap<>();
+        newCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+
+        final List<UpdateAction<Category>> setCustomFieldsUpdateActions =
+            buildSetCustomFieldsUpdateActions(oldCustomFields, newCustomFields, mock(Category.class),
+                new CategoryCustomActionBuilder(),null, category -> null);
+
+        assertThat(setCustomFieldsUpdateActions).isNotNull();
+        assertThat(setCustomFieldsUpdateActions).isNotEmpty();
+        assertThat(setCustomFieldsUpdateActions).hasSize(1);
+        assertThat(setCustomFieldsUpdateActions.get(0)).isInstanceOf(SetCustomField.class);
+    }
+
+    @Test
+    public void buildSetCustomFieldsUpdateActions_WithSameCustomFieldValues_ShouldNotBuildUpdateActions() {
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+        oldCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+        oldCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot"));
+
```

Comment:
assign  "sonNodeFactory.instance.booleanNode(true)" and "JsonNodeFactory.instance.objectNode().put("de", "rot"))" to a variable and reuse it in the newCustomfields

Reply:
5879caa